src/startApp.ts

```diff
@@ -88,13 +99,40 @@ async function loadEntryPointFile(functionAppDirectory: string): Promise<void> {
             }
         } catch (err) {
             const error = ensureErrorType(err);
-            worker.log({
-                message: `Worker was unable to load entry point "${currentFile ? currentFile : entryPointPattern}": ${
-                    error.message
-                }`,
-                level: LogLevel.Warning,
-                logCategory: LogCategory.System,
-            });
+            const newMessage = `Worker was unable to load entry point "${currentFile || entryPointPattern}": ${
+                error.message
+            }`;
+
+            if (shouldBlockOnEntryPointError()) {
+                // This is blocking. The event handler will respond saying it failed
+                error.message = newMessage;
+                throw error;
+            } else {
+                // This is not blocking. We directly log the error, but the event handler will respond saying it succeeded
+                worker.log({
+                    message: newMessage,
+                    level: LogLevel.Error,
+                    logCategory: LogCategory.System,
+                });
+            }
+        }
+    }
+}
+
+function shouldBlockOnEntryPointError(): boolean {
+    if (isNode20Plus()) {
+        // Starting with Node 20, this will always be blocking
+        // https://github.com/Azure/azure-functions-nodejs-worker/issues/697
+        return true;
+    } else {
+        const key = 'FUNCTIONS_NODE_BLOCK_ON_ENTRY_POINT_ERROR';
+        if (isDefined(process.env[key])) {
+            return isEnvironmentVariableSet(process.env[key]);
```

Comment:
I'm not super experienced with regex patterns, so what does `isEnvironmentVariableSet` test exactly? Can this be used in place of the `isDefined(process.env[key])` line entirely?

Reply:
`isDefined` checks if it is `null` or `undefined`

`isEnvironmentVariableSet` handles string values as well. It will return `false` for strings like `0`, `false`, `FaLse`, and return `true` for basically anything else including `1`, `true`, `tRue`, `randomstuff`, `99`

They are not interchangeable. In this case, `isDefined` says

> Did the user add `FUNCTIONS_NODE_BLOCK_ON_ENTRY_POINT_ERROR` to their app settings?

If the answer is yes, we use `isEnvironmentVariableSet` to check

> Does the value of the app setting indicate true or false?